### PR TITLE
Polyfill components

### PR DESCRIPTION
**Pull request by 4e6**
Changelog:
- add: implement `Buffer` by delegating to the `buffer` npm package
- add: create `CryptoPolyfill` component with initial crypto implementation
- refactor: separated polyfill components into `polyfill/timers`, `polyfill/crypto` and `polyfill/websocket`.
- refactor: separated `WebSocket` from `WebSocketPolyfill`. 
  - `WebSocketPolyfill` is an implementation of ProxyExecutable API between JVM and JS and doesn't know about other polyfill components.
  - `WebSocket` performs initialization of all the required polyfills and provides a demo program.

### CODE REVIEW

File: src/main/js/all.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { WebsocketProvider } from './node_modules/y-websocket/dist/y-websocket.cjs'
 import * as Y from './node_modules/yjs/dist/yjs.cjs'
 
+globalThis.Buffer = require('buffer').Buffer;
```
Comment (4e6) [inline src/main/js/all.js:4]: This confirms that we can use the `buffer` npm module.
The proper `Buffer` implementation would be to create a separate js project, pack it, and load it to a polyglot context.

File: pom.xml [hunk 2]
```diff
@@ -137,8 +137,8 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <netbeans.compile.on.save>none</netbeans.compile.on.save>
-        <exec.mainClass>org.apidesign.polyfill.websocket.Main</exec.mainClass>
+        <exec.mainClass>org.apidesign.WebSocket</exec.mainClass>
```
Comment (JaroslavTulach) [inline pom.xml:140]: We cannot use `org.apidesign` - that's too generic package. My website, apidesign contains many projects. We need to differentiate between them by using sub packages.

File: src/main/java/org/apidesign/polyfill/Polyfill.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.apidesign.polyfill;
+
+import org.graalvm.polyglot.Context;
+
+public interface Polyfill {
+
+    void initialize(Context ctx);
```
Comment (JaroslavTulach) [inline src/main/java/org/apidesign/polyfill/Polyfill.java:7]: Isn't this over designed? We pretend there are independent _polyfill_ components, but they are not that much independent.

File: src/main/java/org/apidesign/WebSocket.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.graalvm.polyglot.io.IOAccess;
+
+public class WebSocket {
+
+    private WebSocket() {
+    }
+
+    public static void initializePolyfill(Context ctx, ExecutorService executor) {
+        Polyfill[] components = new Polyfill[]{
+            new TimersPolyfill(executor),
```
Comment (JaroslavTulach) [inline src/main/java/org/apidesign/WebSocket.java:24]: The _polyfill_ components have `initialize` method, but some of them (`TimersPolyfill`) need more to be initialized... hence I call this _over designed abstraction_.

**Review by JaroslavTulach — APPROVED** [review]
I don't like the _restructuralization_. Previously it was clear where the entry point to the whole system is - e.g. `WebSocketPolyfill.initialize` - everything else was package private. Now there is too many packages, too many abstractions, initialization is too complex. I don't like it at all.

On the other hand, the goal of the project is to get the `Y.js` server working on Graal.js. If the current structure makes you happy, @4e6, go on and merge this PR. I believe you have all the permissions to do so.

**Pull request merged**